### DISCUSSION TITLE
How to set the status line background color ? (set the color in the empty middle)

### DISCUSSION THREAD

**Discussion by BigfootN** [nvim-lualine/lualine.nvim#1240] [Q&A] [answered]
Hello,

I am desperately trying to the set the background color of the status line in the middle, where no components are set (between `c` and `x`).

I have tried to call this code (once the `lualine` has been setup and loaded):

```lua
vim.api.nvim_set_hl(0, "StatusLine", {
	force = true,
	bg = "<my color>",
	fg = "<my color>",
})

vim.api.nvim_set_hl(0, "StatusLineNC", {
	force = true,
	bg = "<my color>",
	fg = "<my color>",
})
```

Unfortunately it doesn't seem to do anything. The color is still the same. It's like the colors don't seem to be overriding the colors of the theme.

Thank you very much in advance for any help

**Comment by iton0** [comment, marked as answer]
StatusLine and StatusLineNC are the highlight groups for the default statusline of Neovim. For lualine if you do `:hi lualine<TAB>` it should show all the highlight groups associated with lualine. Hope this helps.

**Reply by BigfootN** [reply]
Thank you for your help :)

**Comment by bro69noscope** [comment]
Dunno if you figured it out but that seems impossible to me too.

EDIT: Ok I got it :grin:

Using lazy .. (I'm also using the distro hence why opts)

```lua
return {
  "nvim-lualine/lualine.nvim",
  opts = function(_, opts)
    opts.options = opts.options or {}

    -- Override specific highlights after the theme is set
    local custom_horizon = require("lualine.themes.horizon")
    local bright_pink = "#FF1493"

    -- Set pink background for all available modes
    custom_horizon.normal.c.bg = bright_pink
    custom_horizon.insert.c.bg = bright_pink
    custom_horizon.visual.c.bg = bright_pink
    custom_horizon.replace.c.bg = bright_pink
    custom_horizon.command.c.bg = bright_pink
    custom_horizon.inactive.c.bg = bright_pink
    if custom_horizon.terminal then
      custom_horizon.terminal.c.bg = bright_pink
    end
    if custom_horizon.select then
      custom_horizon.select.c.bg = bright_pink
    end

    opts.options.theme = custom_horizon

    return opts
  end,
}
```

Set to pink for testing purpose.

So you can load a theme by requiring its module, access the tables, changes their values.
This is probably not gonna work for you out of the box because  I'm using `function(_, opts)`, but with a more typical `config = func()` it should be fine.
